internal/scanner.go

```diff
@@ -38,7 +38,12 @@ func (s Scanner) Scan() error {
 			return err
 		}
 
-		output, err := syft.Encode(result, spdxjson.Format())
+		enc, err := spdxjson.NewFormatEncoderWithConfig(spdxjson.DefaultEncoderConfig())
```

Comment:
With this change it now defaults to SPDX 2.3 (previously 2.2): https://github.com/anchore/syft/blob/8ee209a5ae18bee4db9d845abdc6b2d00b41a8ec/syft/format/internal/spdxutil/versions.go#L7

Reply:
Not sure if we should provide this as an option so user can choose a specific version/format. Maybe a `BUILDKIT_SCAN_FORMAT` that would default to `spdxjson`. If user wants to pin to a specific version of this format it could then set `BUILDKIT_SCAN_FORMAT=spdxjson:2.2`.

@cdupuis @jedevc Thoughts?

Reply:
I think allowing specifying formats would be good IMO, it also opens the route to doing CycloneDX at some point.

I'm not quite sure what the user-interface to this should be though - ideally, we could put it in the `--attestation type=sbom,format=...` or somewhere like that? Or maybe it's more generic, and we should allow passing arbitrary parameters, which we discussed at some point on the buildkit repo.

Reply:
> Or maybe it's more generic, and we should allow passing arbitrary parameters, which we discussed at some point on the buildkit repo.

Yes generic attributes that could be inferred as an env var like `BUILDKIT_SCAN_<name>` and then used with `--attestation type=sbom,<name>=...` on client-side would be handy.